### PR TITLE
Refactor dead if branch into single return flow

### PR DESCRIPTION
**Pull request by Isabek**
Make sure that:

- [x] You have read the [contribution guidelines](https://github.com/pgjdbc/r2dbc-postgresql/blob/main/.github/CONTRIBUTING.adoc).
- [ ] You have created a feature request first to discuss your contribution intent. Please reference the feature request ticket number in the pull request.
- [ ] You use the code formatters provided [here](https://github.com/pgjdbc/r2dbc-postgresql/blob/master/intellij-style.xml) and have them applied to your changes. Don't submit any formatting related changes.
- [ ] You submit test cases (unit or integration tests) that back your changes.

### CODE REVIEW

File: src/main/java/io/r2dbc/postgresql/PostgresqlConnectionFactory.java [hunk 1]
```diff
@@ -94,11 +94,7 @@ private static SocketAddress createSocketAddress(PostgresqlConnectionConfigurati
             return InetSocketAddress.createUnresolved(configuration.getRequiredHost(), configuration.getPort());
         }
 
-        if (configuration.isUseSocket()) {
-            return new DomainSocketAddress(configuration.getRequiredSocket());
-        }
-
-        throw new IllegalArgumentException("Cannot create SocketAddress for " + configuration);
+        return new DomainSocketAddress(configuration.getRequiredSocket());
```
Comment (mp911de) [inline src/main/java/io/r2dbc/postgresql/PostgresqlConnectionFactory.java:97]: When already working on this, we should make this method more robust against absence of `DomainSocketAddress` (that is when someone excludes the epoll transport or native transport in general). Please move the instance creation into an inner static class, something along the lines of:

``` 
    static class Util {

        private static SocketAddress getDomainSocketAddress(PostgresqlConnectionConfiguration configuration) {
            return new DomainSocketAddress(configuration.getRequiredSocket());
        }

    }
```

**Review by mp911de — CHANGES_REQUESTED** [review]

File: src/main/java/io/r2dbc/postgresql/PostgresqlConnectionFactory.java (hunk 1 above)
Comment (Isabek) [reply src/main/java/io/r2dbc/postgresql/PostgresqlConnectionFactory.java:97]: Ok, noted.

Comment (Isabek) [reply src/main/java/io/r2dbc/postgresql/PostgresqlConnectionFactory.java]: @mp911de please, check it again.

**Review by mp911de — APPROVED** [review]

**Comment by mp911de** [comment]
Thank you for your contribution. That's merged, polished, and backported now.

**Pull request closed without merging**